zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -2156,6 +2160,21 @@ public void connect(final long maxMsToWaitUntilConnected, Watcher watcher)
 
       LOG.debug("ZkClient created with _uid {}, _eventThread {}", _uid, _eventThread.getId());
 
+      // initiate monitor
+      try {
+        if (_monitorKey != null && !_monitorKey.isEmpty() && _monitorType != null && !_monitorType
+            .isEmpty()) {
+          _monitor =
+              new ZkClientMonitor(_monitorType, _monitorKey, _monitorInstanceName, _monitorRootPathOnly,
```

Comment:
We should consider closing the monitor to avoid leakage. If the later part of code zk connection timeout and throws exception, the monitor should also be closed.

Reply:
My thinking is that ZkClientMonitor does not hold any session object, nor thread. Let GC deal with it would be fine. Or do you see otherwise?

Reply:
Why we move this into connect()? what problem we try to solve here?